examples/walkthrough.cljs

```diff
@@ -0,0 +1,115 @@
+(ns abio.node.walkthrough
+  (:require [abio.io :as io]
+            [abio.node :as node]
+            [abio.core :refer [*io-bindings*]]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Set up the host specific bindings.
+;;
+;; This is the first function that's called, setting up the host specific I/O
+;; machinery used by the higher level functions.
+(abio.core/set-bindings! (node/bindings))
+
+;;;;;;;;;;;;;;;;;;;;;;
+;; General FS bindings
+;;
+;; These are things that don't, necessarily, warrant opening up some Reader or
+;; Writer, currently limited to listing files, returning the host path
+;; separator, and checking whether some path is a directory
+
+;; In both of these examples we're calling the methods defined in `abio.io`
+;; against the `abio.core` dynamic var `*io-bindings*`, which we'd defined in
+;; our call to `set-bindings!`. Whatever record is stored in `*io-bindings*`
+;; needs to extend the `abio.io/IBindings` protocol.
+
+;; Synchronous directory listing
+(defn sync-ls
+  [path]
+  (if (not (io/-directory? *io-bindings* path))
+    (println "Path needs to be a directory, but you gave me the following: " path)
+    (do (println "Synchronously listing files in " path)
+        (io/-list-files *io-bindings* path))))
+
+;; Asynchronous directory listing
+(defn async-ls-cb
+  [err files]
+  (println "\nHere's the list of files we got back from the asynchronous -list-files call.")
```

Comment:
Maybe this has been already discussed but wouldn't it be better if callbacks returned their argument? Seems like a better default then printing out, like an identity function.

Reply:
Oh I see this is the walkthrough, never mind.

Reply:
Another aspect of this is that the callback runs outside of the...context? for lack of a better word.... of the calling function. So in this case, the only function that returns a value we can work directly with is `async-ls`. If we want to save and work with the data passed to the callback we'd need save it in a var or atom in the namespace. Probably an atom though; defining namespace vars inside of a function seems kind of gauche (and maybe impossible in CLJS? I'm not sure).

Reply:
I so agree with atom thing that this is the hack I employee in inf-clojure for getting completions: https://github.com/clojure-emacs/inf-clojure/blob/master/inf-clojure.el#L910 :smile: